code/modules/vending/vending.dm

```diff
@@ -504,16 +518,220 @@
+		if("vend")
+			if(params["target"])
+				src.vend_ready = 1
+				var/datum/db_record/account = null
+				account = FindBankAccountByName(src.scan?.registered)
+				if ((!src.allowed(usr)) && (!src.emagged) && (src.wires & WIRE_SCANID))
+					boutput(usr, "<span class='alert'>Access denied.</span>") //Unless emagged of course
+					flick(src.icon_deny,src)
+					return
+				if (src.pay)
+					if (src.acceptcard && src.scan)
+						if (!account)
+							boutput(usr, "<span class='alert'>No bank account associated with ID found.</span>")
+							flick(src.icon_deny,src)
+							src.vend_ready = 0
+							src.paying_for = params["target"]
+							return
+						if (account["current_money"] < params["cost"])
+							boutput(usr, "<span class='alert'>Insufficient funds in account. To use machine credit, log out.</span>")
+							flick(src.icon_deny,src)
+							src.vend_ready = 0
+							src.paying_for = params["target"]
+							return
+					else
+						if (src.credit < params["cost"])
+							boutput(usr, "<span class='alert'>Insufficient Credit.</span>")
+							flick(src.icon_deny,src)
+							src.vend_ready = 0
+							src.paying_for = params["target"]
+							return
+ // copy pasted stuff below here
+				SPAWN(src.vend_delay)
+					var/product_amount = 0 // this is to make absolutely sure that these numbers arent desynced
+					var/datum/data/vending_product/product
+
+					var/list/plist
+					if (player_list)
+						plist = player_list
+					else
+						plist = product_list
+					for (var/datum/data/vending_product/R in plist)
+						if(R.product_path == text2path(params["target"]))
+							product_amount = R.product_amount
+							product = R
+					if(!vend_ready) // do not proceed if players dont have money
+						return
+					if(product_amount > 0 && text2path(params["target"]))
+						src.prevend_effect()
+						var/atom/product_path = text2path(params["target"])
+						var/atom/movable/vended = new product_path(src.get_output_location())
+						vended.name = product.product_name
+						vended.loc = src.get_output_location()
+						vended.layer = src.layer + 0.1 //So things stop spawning under the fukin thing
+						if(isitem(vended))
+							usr.put_in_hand_or_eject(vended) // try to eject it into the users hand, if we can
+							src.postvend_effect()
+						if (plist == player_list && product_amount == 1)
+							player_list -= product
+						product.product_amount--
+						if (src.pay && vended) // do we need to take their money
+							if (src.acceptcard && account)
+								account["current_money"] -= product.product_cost
+							else
+								src.credit -= product.product_cost
+						src.vend_ready = 1
+						update_static_data(usr)
+	. = TRUE
+
+/obj/machinery/vending/attack_hand(mob/user as mob)
 	if (status & (BROKEN|NOPOWER))
 		return
-	src.add_dialog(user)
+	//src.add_dialog(user)
 
 	if (src.seconds_electrified != 0)
 		if (src.shock(user, 100))
 			return
 
-	if (!src.HTML)
+	ui_interact(user)
+
+	/*if (!src.HTML)
```

Comment:
well github won't let me suggest removing this, but just delete the old UI code instead of commenting it, everything exists in git history